### PR TITLE
WIP: dynamic provisioning enablement for Ceph RBD

### PR DESCRIPTION
**Pull request by rootfs**
This is WIP.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/19710)

### CODE REVIEW

**Comment by elsonrodriguez** [comment]
@rootfs Want help testing this?

**Comment by rootfs** [comment]
let's wait till storage class framework is done

**Comment by johscheuer** [comment]
@rootfs I know this is WIP. Just on though from my side: is there a possibility to save and access the storage configuration for Ceph in a ConfigMap? Otherwise I would recommend something like this:

```json
{
  "ceph": {
    "Mon": ["10.16.154.78:6789"],
    "Pool": "kube",
    "Id": "kube",
    "Keyring": "/etc/ceph/ceph.client.kube.keyring",
    "FSType": "ext4"
  }
}
```

This would allow to use multiple Storage providers within the same cluster e.q. glusterFS and Ceph (only an idea).

File: examples/experimental/persistent-volume-provisioning/README.md [hunk 1]
```diff
@@ -118,6 +118,77 @@ persistentvolumeclaim "claim1" deleted
...
+```console
+# kubectl create -f claim1.json 
+persistentvolumeclaim "claim1" created
+```
+
+Confirm the PV is created using RBD volume:
+```console
+# kubectl.sh get pv
+NAME           LABELS    CAPACITY   ACCESSMODES   STATUS    CLAIM            REASON    AGE
+pv-rbd-7oiyw   <none>    1Mi        RWO           Bound     default/claim1             12s
```
Comment (MikaelCluseau) [inline examples/experimental/persistent-volume-provisioning/README.md]: Given that namespace cannot contain dots but PV names can, couldn't we name the generated PV "{{.PVC.Namespace}}.{{.PVC.Name}}"?

Comment (MikaelCluseau) [inline examples/experimental/persistent-volume-provisioning/README.md]: or "{{.PVC.Namespace}}.{{.PVC.Name}}-{{.RandomString}}"

**Comment by rootfs** [comment]
@johscheuer the details of storing ceph cluster config is an issue. we'll address it once the DP framework is ready.

**Comment by dchen1107** [comment]
The pr was opened half year ago, and re-assign to me by bot 20 days ago. Is this pr still on-going? Or staled due to some other unresolved dependencies? Please let me know. If it is the former case, I will allocate time to review it. Thanks!

**Comment by rootfs** [comment]
This PR is waiting for #29006 to merge first.

**Comment by rootfs** [comment]
@elsonrodriguez @johscheuer @MikaelCluseau I am closing this one. #30959 is based on new DP framework.

**Pull request closed without merging**